### PR TITLE
Add InstallHistory table

### PR DESCRIPTION
**Pull request by timzimmermann**
See #1922.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by theopolis** [comment]
Simple and clean! @osqueryer, please test this

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
Build origin/pr/1925/merge (Job: 2480) is good! :+1:

**Comment by theopolis** [comment]
We should do a quick review of the column names. If there are any similar names that would function as foreign keys it would be awesome to use those existing column names-- essentially overriding/ignoring the keys in the install history plist.

File: specs/darwin/install_history.table [hunk 1]
```diff
@@ -0,0 +1,11 @@
+table_name("install_history")
+description("OS X package install history.")
+schema([
+    Column("date", INTEGER, "Label date as UNIX timestamp"),
+    Column("display_name", TEXT, "Label displayName"),
```
Comment (theopolis) [inline specs/darwin/install_history.table]: Can we name this `name`?

File: specs/darwin/install_history.table [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    Column("display_version", TEXT, "Label displayVersion"),
```
Comment (theopolis) [inline specs/darwin/install_history.table]: How about just `version`?

File: specs/darwin/install_history.table [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    Column("process_name", TEXT, "Label processName"),
+    Column("package_identifier", TEXT, "Label packageIdentifiers"),
```
Comment (theopolis) [inline specs/darwin/install_history.table]: If we use `package_id` then it's consistent with the `package_receipts` table.

File: specs/darwin/install_history.table [hunk 4]
```diff
@@ -0,0 +1,11 @@
+table_name("install_history")
+description("OS X package install history.")
+schema([
+    Column("date", INTEGER, "Label date as UNIX timestamp"),
+    Column("display_name", TEXT, "Label displayName"),
+    Column("display_version", TEXT, "Label displayVersion"),
+    Column("process_name", TEXT, "Label processName"),
```
Comment (theopolis) [inline specs/darwin/install_history.table]: On my host this is only `installer` or `storedownloadd`, maybe this is more indicative of the package install source or reason-- would `install_source` or `source` be more appropriate?

File: specs/darwin/install_history.table [hunk 5]
```diff
@@ -0,0 +1,11 @@
+table_name("install_history")
```
Comment (theopolis) [inline specs/darwin/install_history.table]: Let's prefix this with `package_` or rename to `package_history`. Then it's easier to discover/relate with `package_receipts` and `package_bom`.

**Comment by theopolis** [comment]
Hey @timzimmermann, any chance you've had a moment to look at the notes above?

**Comment by timzimmermann** [comment]
Honestly, I haven't had a chance for now. I will try to look into it tonight.

**Comment by timzimmermann** [comment]
@theopolis: I am so sorry for not getting back on this, I couldn't find the time. Thanks for refactoring!

**Comment by theopolis** [comment]
No worries at all :D

**Pull request closed without merging**